pkg/client/fileCheckpointer.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2022 IBM All Rights Reserved.
+
+SPDX-License-Identifier: Apache-2.0
+*/
+
+package client
+
+import (
+	"encoding/json"
+	"io/ioutil"
+	"os"
+)
+
+type FileCheckpointer struct {
+	path          string
+	blockNumber   uint64
+	transactionID string
+}
+
+type checkpointState struct {
+	BlockNumber   uint64 `json:"blockNumber"`
+	TransactionID string `json:"transactionId"`
+}
+
+func NewFileCheckpointer(path string) (*FileCheckpointer, error) {
+	fileCheckpointer := new(FileCheckpointer)
+	fileCheckpointer.path = path
+
+	err := fileCheckpointer.loadFromFile()
+	if err != nil {
+		return fileCheckpointer, err
+	}
+
+	err = fileCheckpointer.saveToFile()
+	return fileCheckpointer, err
```

Comment:
We typically wouldn't return a result along with an error. The error indicates we failed to create a result.
```suggestion
	if err := fileCheckpointer.loadFromFile(); err != nil {
		return nil, err
	}

	if err := fileCheckpointer.saveToFile(); err != nil {
		return nil, err
	}
	
	return fileCheckpointer, nil
```